copilot-theorem/src/Copilot/Theorem/What4/Translate.hs

```diff
@@ -240,7 +241,9 @@ data CopilotValue a = CopilotValue { cvType :: CT.Type a
                                    , cvVal :: a
                                    }
 
-valFromExpr :: sym ~ WB.ExprBuilder t st fs =>
+valFromExpr ::
```

Comment:
I had to do a bit of surgery to repair this function:

1. I have to explicitly match on the `FloatModeRepr` in `iFloatGroundEval` to make the `GroundEvalFn` be able to do its thing. It's a bit heavyweight, but it gets the job done.
2. Speaking of which, in order to even know what the `FloatModeRepr` is, I added an additional `KnownRepr WB.FloatModeRepr fm` constraint. This changes the API a bit, but then again, I can't find any call sites for this function, so I think we have some flexibility in changing it.